shell/common/rasterizer.h

```diff
@@ -48,6 +49,54 @@ class AiksContext;
 
 namespace flutter {
 
+// The result status of Rasterizer::Draw. This is only used for unit tests.
+enum class DrawStatus {
+  // The drawing was done without any specified status.
+  kDone,
+  // Failed to rasterize the frame because the Rasterizer is not set up.
+  kNotSetUp,
+  // Nothing was done, because the call was not on the raster thread. Yielded to
+  // let this frame be serviced on the right thread.
+  kYielded,
+  // Nothing was done, because pipeline was empty.
+  kPipelineEmpty,
+  // Nothing was done, because GPU was unavailable.
+  kGpuUnavailable,
+};
+
+// The result status of drawing to a view. This is only used for unit tests.
+enum class DrawSurfaceStatus {
+  // The layer tree was successfully rasterized.
+  kSuccess,
+  // The layer tree should be submitted again.
+  //
+  // This can occur on Android when switching the background surface to
+  // FlutterImageView.  On Android, the first frame doesn't make the image
+  // available to the ImageReader right away. The second frame does.
+  // TODO(egarciad): https://github.com/flutter/flutter/issues/65652
+  //
+  // This can also occur when the frame is dropped to wait for the thread
+  // merger to merge the raster and platform threads.
+  kRetry,
+  // Failed to rasterize the frame.
+  kFailed,
+  // Layer tree was discarded because its size does not match the view size.
+  // This typically occurs during resizing.
+  kDiscarded,
+};
+
+// The information to draw to all views of a frame.
+struct FrameItem {
+  FrameItem(std::list<LayerTreeTask> tasks,
+            std::unique_ptr<FrameTimingsRecorder> frame_timings_recorder)
+      : layer_tree_tasks(std::move(tasks)),
+        frame_timings_recorder(std::move(frame_timings_recorder)) {}
+  std::list<LayerTreeTask> layer_tree_tasks;
```

Comment:
Could you expand on why you chose a `std::list` instead of a `std::vector<std::unique_ptr<T>>`? Generally I'd lean towards the `std::vector` approach.

Reply:
I chose it mostly because it is pushed one after another to avoid reallocating. I'm not hard on it. But why do you think vector is better? And do you specifically recommend a unique_ptr as well?

Reply:
> But why do you think vector is better?

There's trade-offs so I wouldn't say one is always better than the other. The `std::list` has different performance characteristics than `std::vector`, but in this case you're appending/iterating which is ideal for `std::vector`.

> And do you specifically recommend a unique_ptr as well?

More trade-offs, but the benefits are:

1. If appending the `std::vector` requires reallocating the underlying array, `std::unique_ptr` enables the items to be moved instead of copied
2. You can move items in/out of the vector instead of copying (not needed here)
3. Using `std::unique_ptr` avoids issues if items have different sizes (not needed here)